### PR TITLE
D0.1 a 3 artificial objects

### PR DESCRIPTION
**Pull request by hfanahita**
The code for this task is completed. Please let me know if there are any changes required.

### CODE REVIEW

**Review by mehdi138r — CHANGES_REQUESTED** [review]

File: src/simulator/world.py [hunk 1]
```diff
@@ -96,9 +97,19 @@ def intersect(self) -> tuple:
...
                                                     self.registry.get_objects()[oid_2])
-                    # instance has to be added for both objects
-                    intersection_result[oid_1].append(instance)
-                    intersection_result[oid_2].append(instance)
+                    # instance has to be added for intersections with physical objects only
+                    if isinstance(self.registry.get_objects()[oid_1], non_physical_object.NonPhysicalObject):
+                        intersection_result[oid_1].append(instance)
+                    elif isinstance(self.registry.get_objects()[oid_2], non_physical_object.NonPhysicalObject):
+                        intersection_result[oid_2].append(instance)
+                    elif isinstance(self.registry.get_objects()[oid_1],
```
[the remark this answers is not in the document]
Comment (hfanahita) [reply src/simulator/world.py:105]: I don't think so. If we remove this condition, then when two non-physical objects intersect, the intersection instance will be added for both of them, which as we discussed in the last meeting, is not what we want.

[the remark this answers is not in the document]
Comment (erfanz) [reply src/simulator/world.py:105]: @hfanahita 
Your code is essentially doing the follow: 
```
If A { ... }
elif B { ... } 
elif A and B { ... }
```
If both A and B are correct, the first `if` statement will get through, not the third one.

[the remark this answers is not in the document]
Comment (hfanahita) [reply src/simulator/world.py:105]: Oh, sorry. You're right, so basically I should change my code to:
```
if A and B { ... }
elif A { ... }
elif B { ... }
else { ... }
```
Is that correct?

[the remark this answers is not in the document]
Comment (mehdi138r) [reply src/simulator/world.py]: I think now it's correct.

[the remark this answers is not in the document]
Comment (erfanz) [reply src/simulator/world.py]: @hfanahita 
Your change looks good. I think the code will be shorter and more readable if you use short-hand variables for `isinstance(self.registry.get_objects()[oid_X], non_physical_object.NonPhysicalObject)`.

File: src/simulator/world.py [hunk 2]
```diff
@@ -86,19 +87,28 @@ def intersect(self) -> tuple:
...
-                if not (self.registry.get_objects()[oid_1].shape == None \
-                        or self.registry.get_objects()[oid_2].shape == None):
-                    instance = IntersectionInstance(self.registry.get_objects()[oid_1],
-                                                    self.registry.get_objects()[oid_2])
-                    # instance has to be added for both objects
-                    intersection_result[oid_1].append(instance)
-                    intersection_result[oid_2].append(instance)
+                if not (object_1.shape == None or object_2.shape == None):
+                    instance = IntersectionInstance(object_1, object_2)
+                    # instance has to be added for intersections with physical objects only
```
Comment (saeedafra) [inline src/simulator/world.py:101]: it seems that he code is in good shape. Could you please add a TODO here just for reference, saying that this whole situation is temporary and at this stage our only nonphysicalobject is distance measurement beam. Thanks

**Review by saeedafra — CHANGES_REQUESTED** [review]

**Review by erfanz — APPROVED** [review]

**Review by saeedafra — APPROVED** [review]

**Pull request merged**